endpoints/openrtb2/amp_auction.go

```diff
@@ -236,8 +236,14 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		HookExecutor:               deps.hookExecutor,
 	}
 
-	response, err := deps.ex.HoldAuction(ctx, auctionRequest, nil)
+	aucResponse, err := deps.ex.HoldAuction(ctx, auctionRequest, nil)
+	var response *openrtb2.BidResponse
+	if aucResponse != nil {
+		response = aucResponse.BidResponse
+		ao.SeatNonBid = aucResponse.GetSeatNonBid()
+	}
```

Comment:
```go
ao.AuctionResponse = auctionResponse.BidResponse
```
@SyntaxNode :  Just worrying about above code snippet. I believe it will fail in case `auctionResponse` is `nil`.

Other changes can be done. 
1. Removed `response` variable
2. convert receiver of `GetSeatNonBid()` to pointer and added nil check there

Reply:
That's a good point. We could also solve this problem by always returning an AuctionResponse value instead of a pointer, but that may be a large change and there is a pleasantness to returning `nil` when there is an error.

I am good with either leaving this as (skeptically) returning a value instead of a pointer.